zerver/models.py

```diff
@@ -1196,21 +1186,15 @@ def get_realm_emoji_dicts(realm_id: int, only_active_emojis: bool = False) -> Di
     return d
 
 
-def get_realm_emoji_by_id_uncached(realm_id: int) -> Dict[str, EmojiInfo]:
-    return get_realm_emoji_dicts(realm_id)
-
-
 @cache_with_key(get_realm_emoji_by_id_cache_key, timeout=3600 * 24 * 7)
 def get_realm_emoji_by_id(realm_id: int) -> Dict[str, EmojiInfo]:
     return get_realm_emoji_by_id_uncached(realm_id)
 
 
-def get_active_realm_emoji_uncached(realm_id: int) -> Dict[str, EmojiInfo]:
-    realm_emojis = get_realm_emoji_dicts(realm_id, only_active_emojis=True)
-    d = {}
-    for emoji_id, emoji_dict in realm_emojis.items():
-        d[emoji_dict["name"]] = emoji_dict
-    return d
+def get_realm_active_emoji(realm_id: int) -> Dict[str, EmojiInfo]:
+    # It's important to use the cached version here.
+    realm_emojis = get_realm_emoji_by_id(realm_id)
+    return {row["name"]: row for row in realm_emojis.values() if not row["deactivated"]}
```

Comment:
I think in my ideal world, we'd have a single function called by other modules, `get_realm_emoji(realm_id: int, *, include_deactivated: bool)`, so that one can't call `get_realm_emoji_by_id` and accidentally not realize you're incorrectly including deactivated custom emoji that should be ignored ... and that function would, use a single cache that just gets all the custom emoji by the realm and just filters internally the deactivated custom emoji to pass back to its caller.

Thoughts?

Reply:
The hindrance to this suggestion is that when you're dealing with active emojis you want "name" as the key, whereas the other function uses "id" as the key.  I kinda hate boolean args (even when they are explicit, as in your suggestion), and I hate even more to return some kind of Union type.

I'll address this better names.